encodings/dict/src/compute/binary_numeric.rs

```diff
@@ -0,0 +1,29 @@
+use vortex_array::array::ConstantArray;
+use vortex_array::compute::{binary_numeric, BinaryNumericFn};
+use vortex_array::{ArrayData, IntoArrayData};
+use vortex_error::VortexResult;
+use vortex_scalar::BinaryNumericOperator;
+
+use crate::{DictArray, DictEncoding};
+
+impl BinaryNumericFn<DictArray> for DictEncoding {
+    fn binary_numeric(
+        &self,
+        array: &DictArray,
+        rhs: &ArrayData,
+        op: BinaryNumericOperator,
+    ) -> VortexResult<Option<ArrayData>> {
```

Comment:
We need a pre-check to verify that the values of the dictionary array are of a primitive type. If not, we can fast-fail to avoid creating a new ConstantArray.

Reply:
Hmmm, maybe it's okay because we could assume that users should know that and this bad case is rare.